models/lwr/openmc_pincells.py

```diff
@@ -0,0 +1,140 @@
+#--------------------------------------------------------------------------------------------------------------------------#
+# This is based on the C5G7 reactor physics benchmark problem extension case as described in:                              #
+# "Benchmark on Deterministic Transport Calculations Without Spatial Homogenisation: MOX Fuel Assembly 3-D Extension Case" #
+# [NEA/NSC/DOC(2005)16]                                                                                                    #
+# https://www.oecd-nea.org/upload/docs/application/pdf/2019-12/nsc-doc2005-16.pdf                                          #
+#                                                                                                                          #
+# The original C5G7 benchmark is defined with multi-group cross sections. To account for                                   #
+# continuous energy spectral effects, we chose to use the material properties provided in:                                 #
+# "Proposal for a Second Stage of the Benchmark on Power Distributions Within Assemblies"                                  #
+# [NEA/NSC/DOC(96)2]                                                                                                       #
+# https://www.oecd-nea.org/upload/docs/application/pdf/2020-01/nsc-doc96-02-rev2.pdf                                       #
+#--------------------------------------------------------------------------------------------------------------------------#
+
+import openmc
+import numpy as np
+import openmc_common as geom
+import openmc_materials as mats
+
+#--------------------------------------------------------------------------------------------------------------------------#
+# Geometry definitions.
+## The pincells - fuels first
+### Common primitives for defining the different fuel regions.
+fuel_pin_or = openmc.ZCylinder(r = geom.r_fuel)
+fuel_gap_1_or = openmc.ZCylinder(r = geom.r_fuel + geom.t_f_c_gap)
+fuel_zr_or = openmc.ZCylinder(r = geom.r_fuel + geom.t_f_c_gap + geom.t_zr_clad)
+fuel_bb = openmc.model.RectangularPrism(width = geom.pitch, height = geom.pitch)
+
+### The entire 4.3% MOX pincell.
+gap_1_cell_1 = openmc.Cell(name = '4.3% MOX Pin Gap 1')
+gap_1_cell_1.region = +fuel_pin_or & -fuel_gap_1_or
+zr_clad_cell_1 = openmc.Cell(name = '4.3% MOX Pin Zr Clad')
+zr_clad_cell_1.region = +fuel_gap_1_or & -fuel_zr_or
+zr_clad_cell_1.fill = mats.zr
+h2o_bb_cell_1 = openmc.Cell(name = '4.3% MOX Pin Water Bounding Box')
+h2o_bb_cell_1.region = +fuel_zr_or & -fuel_bb
+h2o_bb_cell_1.fill = mats.h2o
+
+mox_4_3_fuel_cell = openmc.Cell(name = '4.3% MOX Fuel Pin')
+mox_4_3_fuel_cell.region = -fuel_pin_or
+mox_4_3_fuel_cell.fill = mats.mox_4_3
+mox43_u = openmc.Universe(cells=[mox_4_3_fuel_cell, gap_1_cell_1, zr_clad_cell_1, h2o_bb_cell_1])
```

Comment:
I would recommend a more semantic data structure:
```
gap_cell['mox_4_3'] = ...
zr_clad_cell['mox_4_3'] = ...
etc...

fuel_cell['mox_4_3'] = ... 
```
(assuming you can't reuse cells that are the same across different universes - see question below)